clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1427,7 +1427,7 @@ private void acquire() {
         ensureNotClosed();
         long threadId = Thread.currentThread().getId();
         if (threadId != currentThread.get() && !currentThread.compareAndSet(NO_CURRENT_THREAD, threadId))
-            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access");
+            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access. Request accessing thread is " + threadId + " and it is already being accessed by " + currentThread.get());
```

Comment:
@bharatviswa504 @ijuma Couldn't we just change the thread ID that we store to an `AtomicReference<Thread>` to store the entire `Thread` object instead of the ID and make `NO_CURRENT_THREAD` `null`? You would just need to be careful to `currentThread.get()` once and once you verify it is non-null you can `getName()` safely. In fact, this is best effort concurrency detection, so it's even ok to call `currentThread.get()` twice as long as we're careful about null values in all cases.